v1/lib/server/readMetadata.js

```diff
@@ -59,46 +59,55 @@ function readSidebar(sidebars = {}) {
   Object.keys(sidebars).forEach(sidebar => {
     const categories = sidebars[sidebar];
 
-    let ids = [];
-    const categoryOrder = [];
-    const subCategoryOrder = [];
+    let sidebarItems = [];
     Object.keys(categories).forEach(category => {
-      if (Array.isArray(categories[category])) {
-        ids = ids.concat(categories[category]);
-
-        for (let i = 0; i < categories[category].length; i++) {
-          categoryOrder.push(category);
-          subCategoryOrder.push(undefined);
+      const categoryItems = categories[category];
+      categoryItems.forEach(categoryItem => {
+        if (typeof categoryItem === 'object') {
+          Object.keys(categoryItem).forEach(subcategory => {
+            const subcategoryItems = categoryItem[subcategory];
+            subcategoryItems.forEach(subcategoryItem => {
+              sidebarItems.push({
+                id: subcategoryItem,
+                category,
+                subcategory,
+                sort: sidebarItems.length + 1,
```

Comment:
This is more of a question.

Do you think we need the `sort`. To be honest im not sure whether we really need it.

Reply:
Hmm true. I'll remove this and see if it still works. This sort is added sequentially and it doesn't seem to do anything.